openedx_external_enrollments/external_enrollments/icc_external_enrollment.py

```diff
@@ -0,0 +1,198 @@
+"""ICCExternalEnrollment class file."""
+import json
+import logging
+from uuid import uuid4
+
+from django.conf import settings
+from rest_framework import status
+import requests
+import xmltodict
+
+from openedx_external_enrollments.external_enrollments.base_external_enrollment import BaseExternalEnrollment
+from openedx_external_enrollments.edxapp_wrapper.get_student import get_user
+from openedx_external_enrollments.models import EnrollmentRequestLog
+
+LOG = logging.getLogger(__name__)
+
+
+class ICCExternalEnrollment(BaseExternalEnrollment):
+    """
+    ICCExternalEnrollment class.
+    """
+    USER_CREATION_MAX_RETRIES = 5
+    USERNAME_SUFFIX_LENGTH = 5
+
+    def __str__(self):
+        return "ICC"
+
+    def _execute_post(self, url, data=None, headers=None, json_data=None):
+        """
+        Execute post request to achieve the ICC external enrollment.
+        """
+        response = requests.post(
+            url=url,
+            data=json_data,
+        )
+
+        return response
+
+    def _get_enrollment_headers(self):
+        """
+        Method that returns None by default, ICC integration does not headers.
+        """
+        return None
```

Comment:
It's no necessary to return "None" explicitly, please remove it. @ivanvgh

Reply:
In this case is necessary because '_post_enrollment' method from 'BaseExternalEnrollment' needs it explicitly.

Reply:
Where? Could you point the line? @ivanvgh

Reply:
SUre, it is in 'openedx_external_enrollments/external_enrollments/base_external_enrollment.py'  line 47. If we don't implement it it will thrown NotImplementedError from line 87.